tests/test_compiler.py

```diff
@@ -0,0 +1,346 @@
+        """Test compiler active and available methods"""
+
+        assert not qml.compiler.active()
+
+        assert qml.compiler.available("catalyst")
+        assert qml.compiler.available_compilers() == ["catalyst"]
+
+        assert qml.compiler.available("catalyst")
+        assert qml.compiler.available_compilers() == ["catalyst"]
+
+    def test_active_compiler(self):
+        """Test `qml.compiler.active_compiler` inside a simple circuit"""
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(phi, theta):
+            if qml.compiler.active_compiler() == "catalyst":
+                qml.RX(phi, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(theta, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        assert jnp.allclose(circuit(jnp.pi, jnp.pi / 2), 1.0)
+        assert jnp.allclose(qml.qjit(circuit)(jnp.pi, jnp.pi / 2), -1.0)
+
+    def test_active(self):
+        """Test `qml.compiler.active` inside a simple circuit"""
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(phi, theta):
+            if qml.compiler.active():
+                qml.RX(phi, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(theta, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        assert jnp.allclose(circuit(jnp.pi, jnp.pi / 2), 1.0)
+        assert jnp.allclose(qml.qjit(circuit)(jnp.pi, jnp.pi / 2), -1.0)
+
+    def test_qjit_circuit(self):
+        """Test JIT compilation of a circuit with 2-qubit"""
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qjit
+        @qml.qnode(dev)
+        def circuit(theta):
+            qml.Hadamard(wires=0)
+            qml.RX(theta, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(wires=1))
+
+        assert jnp.allclose(circuit(0.5), 0.0)
+
+    def test_qjit_aot(self):
+        """Test AOT compilation of a circuit with 2-qubit"""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qjit  # compilation happens at definition
+        @qml.qnode(dev)
+        def circuit(x: complex, z: ShapedArray(shape=(3,), dtype=jnp.float64)):
+            theta = jnp.abs(x)
+            qml.RY(theta, wires=0)
+            qml.Rot(z[0], z[1], z[2], wires=0)
+            return qml.state()
+
+        result = circuit(0.2j, jnp.array([0.3, 0.6, 0.9]))
+        expected = jnp.array(
+            [0.75634905 - 0.52801002j, 0.0 + 0.0j, 0.35962678 + 0.14074839j, 0.0 + 0.0j]
+        )
+        assert jnp.allclose(result, expected)
+
+    @pytest.mark.parametrize(
+        "_in,_out",
+        [
+            (0, False),
+            (1, True),
+        ],
+    )
+    def test_variable_capture_multiple_devices(self, _in, _out):
+        """Test variable capture using multiple backend devices."""
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qjit()
```

Comment:
just wanna double-check that we're ok with users specifying `@qml.qjit()` instead of `@qml.qjit`, because we're doing it here (and it works)

Reply:
It is available (because there are optional keyword arguments), but not necessary. We should be consistent and always omit the optional parentheses in our docs.